json/src/spec/authority_round.rs

```diff
@@ -105,6 +107,17 @@ pub struct AuthorityRound {
 	pub params: AuthorityRoundParams,
 }
 
+/// Either a single `T` value, or a map assigning block numbers to transitions to different values of type `T`.
+#[derive(Debug, PartialEq, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub enum TransitionMap<T> {
```

Comment:
As this is a new feature I wonder if we really need to be this fancy. Can't we mandate a map here and avoid this?

Reply:
The [`posdao-contracts` chainspec template](https://github.com/poanetwork/posdao-contracts/blob/6538a3c2e5bbb09f41cf1fb7c28b2405ed35ee8c/templates/spec.json#L19) is an example for how we are currently using this.
If we allow not using a map, it looks just like e.g. the block reward contract.

Reply:
Yes, we suppose that the address of randomness contract can be changed in spec, so we decided to have that as a map, e.g.:

```
"randomnessContractAddress": {
  "0": "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "100": "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
}
```

But we also suggest to have a simpler form of first address definition, e.g. when we define it as

```
"randomnessContractAddress": {
  "0": "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
}
```

we could write that as

```
"randomnessContractAddress": "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

Reply:
I agree that it'd be nice to let knowledgeable users to have a short-form for the single-item use case. However I also think that it's worth more to have less rust code to maintain: configuration mistakes are non-catastrophic (the node won't start; bugs in rust might be more subtle) and probably easy to mitigate with docs. So yeah, unless you disagree strongly I think it's good to mandate a map here.

Reply:
This is addressed in https://github.com/poanetwork/parity-ethereum/commit/4799aeb3d11bac2910ff04e5714631a541d92aec.